src/core/init/get_current_discontinuity_end.ts

```diff
@@ -62,25 +61,24 @@ export default function getDiscontinuities(
       // case of small discontinuity in the content.
       } else if (nextBufferRangeGap < BUFFER_DISCONTINUITY_THRESHOLD) {
         const seekTo = (position + nextBufferRangeGap + 1 / 60);
-        return [position, seekTo];
+        return seekTo;
       }
 
       // 3. Is it a discontinuity between periods ? -> Seek at the beginning of the
       //                                               next period
       const currentPeriod = manifest.getPeriodForTime(position);
-      if (currentPeriod != null) {
+      if (currentPeriod !== undefined) {
         const nextPeriod = manifest.getPeriodAfter(currentPeriod);
-        if (currentPeriod != null &&
-            currentPeriod.end != null &&
-            nextPeriod != null &&
+        if (currentPeriod.end !== undefined &&
+            nextPeriod !== undefined &&
             position > (currentPeriod.end - 1) &&
             position <= nextPeriod.start &&
-            nextPeriod.start - currentPeriod.end === 0) {
-          return [currentPeriod.end, nextPeriod.start];
+            (nextPeriod.start - currentPeriod.end) < 0.5) {
```

Comment:
I don't understand the old code here

Reply:
`nextPeriod.start - currentPeriod.end === 0` looks like a confusing way to say `nextPeriod.start === currentPeriod.end`

Reply:
Yes indeed